src/_griffe/docstrings/google.py

```diff
@@ -491,22 +548,16 @@ def _read_returns_section(
                     annotation = docstring.parent.annotation  # type: ignore[union-attr]
                 else:
                     raise ValueError
-                if len(block) > 1:
-                    if annotation.is_tuple:
-                        annotation = annotation.slice.elements[index]
-                    else:
-                        if annotation.is_iterator:
-                            return_item = annotation.slice
-                        elif annotation.is_generator:
-                            return_item = annotation.slice.elements[2]
-                        else:
-                            raise ValueError
-                        if isinstance(return_item, ExprName):
-                            annotation = return_item
-                        elif return_item.is_tuple:
-                            annotation = return_item.slice.elements[index]
-                        else:
-                            annotation = return_item
+                orig_annotation = annotation
+                annotation = _maybe_destructure_annotation(
+                    annotation, index, multiple=returns_multiple_items
+                )
+                if annotation == orig_annotation:
+                    annotation = _maybe_destructure_annotation(
+                        _unpack_generators(annotation, 2),
+                        index,
+                        multiple=returns_multiple_items,
+                    )
```

Comment:
It all looks good, except maybe this part. I find it a bit hard to follow, compared to the previous version.

Since `_unpack_generators` is called without `mandatory=True` here, couldn't we directly do this?

```python
                annotation = _maybe_destructure_annotation(
                    _unpack_generators(annotation, 2),
                    index,
                    multiple=returns_multiple_items,
                )
```

I'll try this locally and see if tests pass.